src/realm/object-store/sync/sync_manager.cpp

```diff
@@ -262,29 +271,34 @@ void SyncManager::set_log_level(util::Logger::Level level) noexcept
 {
     util::CheckedLockGuard lock(m_mutex);
     m_config.log_level = level;
+    // Update the level threshold in the already created logger
+    if (m_logger_ptr) {
+        m_logger_ptr->set_level_threshold(level);
+    }
 }
 
-void SyncManager::set_logger_factory(SyncClientConfig::LoggerFactory factory) noexcept
+void SyncManager::set_logger_factory(SyncClientConfig::LoggerFactory factory)
 {
     util::CheckedLockGuard lock(m_mutex);
+    if (m_sync_client)
+        throw std::logic_error("Cannot set the logger_factory after creating the sync client");
+
     m_config.logger_factory = std::move(factory);
+    // Replace the already created logger with the new one
+    if (m_config.logger_factory) {
+        m_logger_ptr = m_config.logger_factory(m_config.log_level); // Throws
+    }
 }
 
-std::unique_ptr<util::Logger> SyncManager::make_logger() const
+const std::shared_ptr<util::Logger>& SyncManager::make_logger() const
 {
     util::CheckedLockGuard lock(m_mutex);
     return do_make_logger();
 }
 
-std::unique_ptr<util::Logger> SyncManager::do_make_logger() const
+const std::shared_ptr<util::Logger>& SyncManager::do_make_logger() const
```

Comment:
we should get rid off this

Reply:
Yeah - these functions don't really do what they used to. I thought about changing it to `get_logger()` since that's more descriptive.